### DISCUSSION TITLE
Enable TINYTEXT, TEXT options in Standard field advanced options (as Directus 8 does)

### DISCUSSION THREAD

**Discussion by lluishi93** [directus/directus#4308] [Feature Requests]
It'll be nice to have this possibility back, since some collections do have a lot of string fields (like general label collections), and row size table gets exceeded at some point because all of the fields being created as VARCHAR.

Directus 9
![image](https://user-images.githubusercontent.com/14786440/109304979-59cedf00-783d-11eb-81ae-0236f77260a1.png)

Directus 8
![image](https://user-images.githubusercontent.com/14786440/109305186-a87c7900-783d-11eb-81ab-35ba81388689.png)

**Comment by benhaynes** [comment]
So, for this I assume we would need the full datatype mapping for each database vendor (eg: what are the exact supported datatypes in MySQL vs SQLite). Then we would _keep_ this dropdown as is, but have an `Other` option at the bottom... or an advanced button somewhere, that would show the exact datatypes available for your vendor.

The main issue here is that this might kinda make your abstracted schema (when we have that in YAML/JSON) be a bit tied to the database type. Although we could _set_ the vendor type in the database, and save the more agnostic type in the Directus schema file.

I'm sure @rijkvanzanten will have strong opinions on this one. 😉

**Reply by rijkvanzanten** [reply]
Yeah... that'll be near impossible to make cross-platform 🤔 

To be fair, Directus should already work if you go straight to the database and use a vendor-specific type, so we theoretically could open up an "other" where you can manually type what to use, but for the sake of exporting etc, we'll still have to "dumb it down" to the nearest common denominator I suppose

**Reply by nickrum** [reply]
Doesn't Knex support at least some level of abstraction over different text datatypes?

**Reply by rijkvanzanten** [reply]
That's what we're using now, which is why Directus only surfaces "string" and "text", as those map to VARCHAR/TEXT (or the vendor-equivalent).

The problem here becomes if you want to use a MySQL specific type. Lets say you use `TINYTEXT`, how would we export that in a db-agnostic way? Would we export "fits under the `text` abstraction, "real" type is `TINYTEXT`) and then attempt to use `TINYTEXT` in the import environment and fallback to generic `text` if it crashes? 🤔

**Reply by nickrum** [reply]
I think Knex' text type has an optional parameter to specify the MySQL text datatype preference which should gracefully fallback to using `text` for the other vendors. The "real" type would propably have to be an advanced option which is a no-op on all vendors without support for different types.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
